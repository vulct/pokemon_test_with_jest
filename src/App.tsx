import React, { FC, useEffect, useState } from "react";
import "./App.css";

import axios from "axios";
import PokemonCollection from "./components/PokemonCollection";
import { Pokemon } from "./interface";

interface Pokemons {
  name: string;
  url: string;
}

const App: FC = () => {
  const [pokemons, setPokemons] = useState<Pokemon[]>([]);

  const [nextUrl, setNextUrl] = useState<string>(
    "https://pokeapi.co/api/v2/pokemon?limit=20&offset=20"
  );

  const [loading, setLoading] = useState<boolean>(true);

  useEffect(() => {
    const getPokemon = async () => {
      setLoading(true);

      const res = await axios.get(nextUrl);

      res.data.results.forEach(async (pokemon: Pokemons) => {
        const poke = await axios.get(
          `https://pokeapi.co/api/v2/pokemon/${pokemon.name}`
        );

        setPokemons((p) => [...p, poke.data]);
        setLoading(false);
      });
    };

    getPokemon();
  }, [nextUrl]);

  const handleNextPage = async () => {
    let res = await axios.get(nextUrl);
    setNextUrl(res.data.next);
  };

  return (
    <div className="App">
      <div className="container">
        <div className="pokemon-header">Pokemon</div>
        <PokemonCollection pokemons={pokemons} />
        <div className="btn">
          <button className="loadmore" onClick={handleNextPage}>
            {loading ? "Loading" : "Load more"}
          </button>
        </div>
      </div>
    </div>
  );
};

export default App;
