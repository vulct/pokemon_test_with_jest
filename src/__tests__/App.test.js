import { shallow } from "enzyme";
import App from "../App";

let wrapper;

beforeEach(() => {
  wrapper = shallow(<App />);
});

describe("<App />", () => {
  it("renders one <PokemonCollection /> component", () => {
    expect(wrapper.find("PokemonCollection")).toHaveLength(1);
  });

  it("renders header App component", () => {
    const header = <div className="pokemon-header">Pokemon</div>;
    expect(wrapper.contains(header)).toBe(true);
  });

  describe("show button", () => {
    it("renders button", () => {
      expect(wrapper.find(".btn")).toHaveLength(1);
    });

    it("calls onClick callback", () => {
      wrapper.find(".loadmore").simulate("click");
      expect(wrapper.find(".loadmore").text()).toBe("Loading");
    });
  });
});

test("snapshot App component", () => {
  expect(wrapper).toMatchSnapshot();
});
