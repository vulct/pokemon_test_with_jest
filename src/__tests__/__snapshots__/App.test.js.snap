// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot App component 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <div
      className="pokemon-header"
    >
      Pokemon
    </div>
    <PokemonCollection
      pokemons={Array []}
    />
    <div
      className="btn"
    >
      <button
        className="loadmore"
        onClick={[Function]}
      >
        Loading
      </button>
    </div>
  </div>
</div>
`;
